ui/StatusQ/src/StatusQ/Components/StatusQrCodeScanner.qml

```diff
@@ -7,42 +7,112 @@ import QtGraphicalEffects 1.0
 
 import StatusQ.Controls 0.1
 import StatusQ.Core 0.1
+import StatusQ.Core.Backpressure 1.0
 import StatusQ.Core.Theme 0.1
 
 import QZXing 3.3
 
+/*
+    NOTE:   I'm doing some crazy workarounds here. Tested on MacOS.
+            What I wanted to achieve:
+
+            1. User only gets a OS "allow camera access" popup
+               when a page with QR code scanner is opened.
+            2. Mimize UI freezes, or at least make it obvious
+               that something is going on.
+
+    Camera component uses main UI thread to request OS for available devices.
+    Therefore, we can't simply use Loader with `asyncronous` flag.
+    Neiter we can set `loading: loader.status === Loader.Loading` to this button.
+
+    To achieve desired points, I manually set `loading` property of the button
```

Comment:
This can be mitigated by using Animator: https://github.com/status-im/status-desktop/pull/10012